components/script/dom/browsingcontext.rs

```diff
@@ -84,20 +78,12 @@ impl BrowsingContext {
         }
     }
 
-    pub fn set_active_document(&self, document: &Document) {
-        self.active_document.set(Some(document))
+    pub fn discard(&self) {
+        self.discarded.set(true);
     }
 
-    pub fn active_document(&self) -> Root<Document> {
```

Comment:
We never needed it, so I removed it! SpiderMonkey does the dance to resolve `WindowProxy` to `Window`, which tracks the current document.